src/plugins/discover/public/application/main/components/sidebar/discover_sidebar.tsx

```diff
@@ -325,6 +220,102 @@ export function DiscoverSidebarComponent({
+  const onFilterField: GroupedFieldsParams<DataViewField>['onFilterField'] = useCallback(
+    (field) => {
+      return doesFieldMatchFilters(field, fieldFilter);
+    },
+    [fieldFilter]
+  );
+  const onSupportedFieldFilter: GroupedFieldsParams<DataViewField>['onSupportedFieldFilter'] =
+    useCallback(
+      (field) => {
+        return shouldShowField(field, useNewFieldsApi, isPlainRecord);
+      },
+      [useNewFieldsApi, isPlainRecord]
+    );
+  const onOverrideFieldGroupDetails: GroupedFieldsParams<DataViewField>['onOverrideFieldGroupDetails'] =
+    useCallback((groupName) => {
+      if (groupName === FieldsGroupNames.AvailableFields) {
+        return {
+          helpText: i18n.translate('discover.fieldChooser.availableFieldsTooltip', {
+            defaultMessage:
+              'Your query returned values for these fields. Click + to add an available field to the data table.',
+          }),
+        };
+      }
+      if (groupName === FieldsGroupNames.PopularFields) {
+        return {
+          helpText: i18n.translate('discover.fieldChooser.popularFieldsTooltip', {
+            defaultMessage: 'Add fields to the table that your organization frequently uses.',
+          }),
+        };
+      }
+      if (groupName === FieldsGroupNames.UnmappedFields) {
+        return {
+          helpText: i18n.translate('discover.fieldChooser.unmappedFieldsTooltip', {
+            defaultMessage:
+              "Add fields to the table that aren't explicitly mapped to a field data type.",
+          }),
+        };
+      }
+    }, []);
+  const fieldsExistenceReader = useExistingFieldsReader();
+  const { fieldGroups, scrollToTopResetCounter } = useGroupedFields({
+    dataViewId: (!isPlainRecord && selectedDataView?.id) || null,
+    fieldsExistenceReader: !isPlainRecord ? fieldsExistenceReader : undefined,
+    allFields: allFields || EMPTY_FIELD_LIST,
+    popularFieldsLimit: !isPlainRecord ? popularFieldsLimit : 0,
+    sortedSelectedFields: selectedFields,
+    isAffectedByGlobalFilter: isGlobalFilterApplied,
+    services: {
+      dataViews,
+    },
+    onFilterField,
+    onSupportedFieldFilter,
+    onOverrideFieldGroupDetails,
+  });
+
+  const renderFieldItem: FieldListGroupedProps<DataViewField>['renderFieldItem'] = useCallback(
+    ({ field, groupName }) => (
+      <li key={`field${field.name}`} data-attr-field={field.name}>
+        <DiscoverField
+          alwaysShowActionButton={alwaysShowActionButtons}
+          field={field}
+          highlight={fieldFilter.name}
+          dataView={selectedDataView!}
+          onAddField={onAddField}
+          onRemoveField={onRemoveField}
+          onAddFilter={onAddFilter}
+          documents$={documents$}
+          trackUiMetric={trackUiMetric}
+          multiFields={multiFieldsMap?.get(field.name)} // ideally we better calculate multifields when they are requested first from the popover
```

Comment:
Yes, 100% would improve performance when there are lots of fields ... could
Refactoring multifields could be a nice follow up issue for cleanups ... for now it's fine